File: 8vim/src/main/java/inc/flide/vim8/ui/AboutUsActivity.java
package inc.flide.vim8.ui;

import android.content.Intent;
import android.graphics.Color;
import android.net.Uri;
import android.os.Bundle;
import android.view.MotionEvent;
import android.widget.ImageView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import inc.flide.vim8.R;

public class AboutUsActivity extends AppCompatActivity {
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.about_us_activity_layout);

        setupLink(R.id.constraintLayout_github, "https://github.com/flide/8vim");
        setupLink(R.id.constraintLayout_matrix, "https://app.element.io/#/room/#8vim/lobby:matrix.org");
        setupLink(R.id.constraintLayout_website, "https://8vim.com");
        setupLink(R.id.constraintLayout_twitter, "https://twitter.com/8vim_?s=09");
        setupLink(R.id.constraintLayout_playstore, "https://play.google.com/store/apps/details?id=inc.flide.vi8");
        setupBackArrow();

    }

    private void setupBackArrow() {
        ImageView backArrow = findViewById(R.id.back_arrow);

        backArrow.setOnTouchListener((view, motionEvent) -> {
            if (motionEvent.getAction() == MotionEvent.ACTION_DOWN) {
                backArrow.setBackgroundColor(Color.LTGRAY);
            }
            if (motionEvent.getAction() == MotionEvent.ACTION_UP) {
                backArrow.setBackgroundColor(Color.WHITE);
                Intent intent = new Intent(AboutUsActivity.this, SettingsActivity.class);
                startActivity(intent);
            }
            return true;
        });
    }

    private void setupLink(int constraintLayoutId, String uriString) {
        ConstraintLayout constraintLayout = findViewById(constraintLayoutId);
        constraintLayout.setOnTouchListener((v, event) -> {
            switch (event.getAction()) {
                case MotionEvent.ACTION_DOWN:
                    constraintLayout.setBackgroundColor(Color.LTGRAY);
                    break;
                case MotionEvent.ACTION_UP:
                    constraintLayout.setBackgroundColor(Color.WHITE);
                    Uri uri = Uri.parse(uriString);
                    Intent intent = new Intent(Intent.ACTION_VIEW, uri);
                    startActivity(intent);
                    break;
                default:
            }

            return true;
        });
    }

}

File: 8vim/src/main/java/inc/flide/vim8/ui/AboutUsActivity.kt
package inc.flide.vim8.ui

import android.content.Intent
import android.graphics.*
import android.net.Uri
import android.os.Bundle
import android.view.*
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout
import inc.flide.vim8.R

class AboutUsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.about_us_activity_layout)
        val github: ConstraintLayout = findViewById(R.id.constraintLayout_github)
        github.setOnTouchListener { _: View?, event: MotionEvent ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> github.setBackgroundColor(Color.LTGRAY)
                MotionEvent.ACTION_UP -> {
                    github.setBackgroundColor(Color.WHITE)
                    val uri = Uri.parse("https://github.com/flide/8vim")
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    startActivity(intent)
                }
            }
            true
        }
        val matrix: ConstraintLayout = findViewById(R.id.constraintLayout_matrix)
        matrix.setOnTouchListener { _: View?, event: MotionEvent ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> matrix.setBackgroundColor(Color.LTGRAY)
                MotionEvent.ACTION_UP -> {
                    matrix.setBackgroundColor(Color.WHITE)
                    val uri = Uri.parse("https://app.element.io/#/room/#8vim/lobby:matrix.org")
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    startActivity(intent)
                }
            }
            true
        }
        val website: ConstraintLayout = findViewById(R.id.constraintLayout_website)
        website.setOnTouchListener { _: View?, event: MotionEvent ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> website.setBackgroundColor(Color.LTGRAY)
                MotionEvent.ACTION_UP -> {
                    website.setBackgroundColor(Color.WHITE)
                    val uri = Uri.parse("https://8vim.com")
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    startActivity(intent)
                }
            }
            true
        }
        val twitter: ConstraintLayout = findViewById(R.id.constraintLayout_twitter)
        twitter.setOnTouchListener { _: View?, event: MotionEvent ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> twitter.setBackgroundColor(Color.LTGRAY)
                MotionEvent.ACTION_UP -> {
                    twitter.setBackgroundColor(Color.WHITE)
                    val uri = Uri.parse("https://twitter.com/8vim_?s=09")
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    startActivity(intent)
                }
            }
            true
        }
        val googlePlayStore: ConstraintLayout = findViewById(R.id.constraintLayout_playstore)
        googlePlayStore.setOnTouchListener { _: View?, event: MotionEvent ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> googlePlayStore.setBackgroundColor(Color.LTGRAY)
                MotionEvent.ACTION_UP -> {
                    googlePlayStore.setBackgroundColor(Color.WHITE)
                    val uri =
                        Uri.parse("https://play.google.com/store/apps/details?id=inc.flide.vi8") // missing 'http://' will cause crashed
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    startActivity(intent)
                }
            }
            true
        }
        val backArrow = findViewById<ImageView?>(R.id.back_arrow)
        backArrow.setOnTouchListener { _: View?, motionEvent: MotionEvent ->
            if (motionEvent.action == MotionEvent.ACTION_DOWN) {
                backArrow.setBackgroundColor(Color.LTGRAY)
            }
            if (motionEvent.action == MotionEvent.ACTION_UP) {
                backArrow.setBackgroundColor(Color.WHITE)
                val intent = Intent(this@AboutUsActivity, SettingsActivity::class.java)
                startActivity(intent)
            }
            true
        }
    }
}